File: scripts/analyze_pdf.py

import os
import glob
from pypdf import PdfReader

def main():
    # Find the PDF file
    files = glob.glob(os.path.join(os.getcwd(), "BKGR Lettre Quotidienne*.pdf"))
    if not files:
        print("No matching PDF file found.")
        return

    pdf_path = files[0]
    print(f"Reading PDF: {pdf_path}")
    
    try:
        reader = PdfReader(pdf_path)
        print(f"Number of Pages: {len(reader.pages)}")
        
        text = ""
        for i, page in enumerate(reader.pages):
            page_text = page.extract_text()
            if page_text:
                text += f"\n--- Page {i+1} ---\n"
                text += page_text
        
        output_path = os.path.join(os.getcwd(), "latest_pdf_content.txt")
        print(f"Writing to: {output_path}")
        
        with open(output_path, "w", encoding="utf-8") as f:
            f.write(text)
            
        print(f"Success! Extracted {len(text)} characters.")
        print("--- First 500 characters ---")
        print(text[:500])
        print("----------------------------")
        
    except Exception as e:
        print(f"Error parsing PDF: {e}")

if __name__ == "__main__":
    main()

File: scripts/analyze_pdf.ts

import fs from 'fs';
import path from 'path';
import { createRequire } from 'module';

const require = createRequire(import.meta.url);
const pdf = require('pdf-parse');

const main = async () => {
    try {
        const dir = process.cwd();
        console.log(`Current directory: ${dir}`);

        const files = fs.readdirSync(dir);
        const pdfFile = files.find(f => f.startsWith('BKGR Lettre Quotidienne') && f.endsWith('.pdf'));

        if (!pdfFile) {
            console.error('No matching PDF file found.');
            process.exit(1);
        }

        const pdfPath = path.join(dir, pdfFile);
        console.log(`Reading PDF: ${pdfPath}`);

        if (!fs.existsSync(pdfPath)) {
            console.error(`File does not exist: ${pdfPath}`);
            process.exit(1);
        }

        const dataBuffer = fs.readFileSync(pdfPath);
        console.log(`Read ${dataBuffer.length} bytes.`);

        const data = await pdf(dataBuffer);
        console.log(`Pages: ${data.numpages}`);

        // Save extracted text
        const outputPath = path.join(dir, 'latest_pdf_content.txt');
        console.log(`Writing output to: ${outputPath}`);
        fs.writeFileSync(outputPath, data.text);

        if (fs.existsSync(outputPath)) {
            console.log(`Success! Extracted text saved to ${outputPath}`);
        } else {
            console.error(`Failed to write file at ${outputPath}`);
        }

    } catch (error) {
        console.error('Error in main execution:', error);
        process.exit(1);
    }
};

main();
